File: Divide an Array Into Subarrays With Minimum Cost I/Divide_an_Array_Into_Subarrays_With_Minimum_Cost_I.py

from typing import List
class Solution:
    def minimumCost(self, nums: List[int]) -> int:
        ans = nums[0]
        min1 = float("inf")
        min2 = float("inf")

        for i in range(1,len(nums)):
            curr = nums[i]

            if (curr < min1):
                min2 = min1
                min1 = curr
            elif (curr < min2):
                min2 = curr

        return ans + min1 + min2        

# Test cases based on the examples provided
s = Solution()
inputs = [[1,2,3,12],[5,4,3],[10,3,1,1]]
for nums in inputs:
    print(f"Input: nums = {nums}")
    print(f"Output: {s.minimumCost(nums)}\n")

File: Divide an Array Into Subarrays With Minimum Cost I/Divide_an_Array_Into_Subarrays_With_Minimum_Cost_I.cpp
#include <vector>
#include <algorithm>
#include <iostream>    
using namespace std;

class Solution {
public:
    int minimumCost(vector<int>& nums) {

        sort(nums.begin()+1,nums.end());

        return nums[0] + nums[1] + nums[2];
        

        
    }
};


// Test cases based on the examples provided

int main() {
    Solution s;
    vector<vector<int>> inputs = {{1,2,3,12},{5,4,3},{10,3,1,1}};
    for (const auto& nums : inputs) {
        int result = s.minimumCost(const_cast<vector<int>&>(nums));
        // Print the result
        cout << "Input: nums = [";
        for (size_t i = 0; i < nums.size(); ++i) {
            cout << nums[i];
            if (i < nums.size() - 1) cout << ",";
        }
        cout << "]\n";
        cout << "Output: " << result << "\n\n";
    }
    return 0;
}
